x/profiles/keeper/grpc_query.go

```diff
@@ -122,6 +123,43 @@ func (k Keeper) ChainLinks(ctx context.Context, request *types.QueryChainLinksRe
 	return &types.QueryChainLinksResponse{Links: links, Pagination: pageRes}, nil
 }
 
+// ChainLinkOwners implements the Query/ChainLinkOwners gRPC method
+func (k Keeper) ChainLinkOwners(ctx context.Context, request *types.QueryChainLinkOwnersRequest) (*types.QueryChainLinkOwnersResponse, error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	store := sdkCtx.KVStore(k.storeKey)
+
+	ownersPrefix := types.ChainLinkChainPrefix
+	switch {
```

Comment:
If should have a default case to return error to avoid the case whose fields are empty.

Reply:
The default case is the one before the switch, where the entire `ChainLinkChainPrefix` it's used. I've made it like that because this way it's possible to iterate over all the chain link prefixes. This shouldn't be a problem because there is the pagination anyway.